### PR TITLE
chore: add prow OWNERS files for critial configuration files

### LINKED ISSUE #17004: Strengthen configuration change approval
# Strengthen configuration change approval 
An approval process for the change to the file path scope of global variables and configurations will be deployed.

## Why

The current reviewers and approvers are at the [code repository level](https://github.com/tikv/community/blob/master/teams/tikv/team.json) and are not subdivided into specific areas, can not fine-grained control over changes in important areas. Configuration/variable changes can have a wide-ranging impact, affecting system stability and compatibility. Recently, there have been several online issues caused by configuration/variable changes. 
To minimize the negative impact of configuration/variable changes, ensure that each change undergoes careful consideration and evaluation and comes with clear documentation, and reduce the risk of online issues, we plan to pilot pre-approval of configuration changes in the `tikv/tikv` repo and delegate the approval responsibility to the person in charge of maintaining the TiDB product.

## What's new for developer

The reviewing control framework will be migrated to the native "lgtm" and "approve" plug-ins. In terms of final approval, it will change from a single merge approval (a "/merge" command) to an approval mode by responsible area (one or more "/ approve" command).
All the changes need to be approved by committers or maintainers as before except if there are configurations changes, the PRs need also to be approved by approvers inside `/OWNERS_ALIASES` file. The approvers will be automatically recommended by bot on GitHub.

Ref:
- Backgroup: https://internals.tidb.io/t/topic/854
- Ref: https://github.com/ti-community-infra/configs/issues/824

## Set up

1. Now we use prow's `OWNERS` mechanism to control the approving for pull requests, here is the [PR review flow](https://book.prow.tidb.net/#/en/workflows/pr).
2. We setup an `OWNERS_ALIASES` file in root folder in repository to **maintain the approver teams** for variable or configuration changes.
3. We setup serval `OWNERS` files in sub folders to support approving by folder or file paths. And make it not inherit from parent `OWNERS` files to ensure the changes must be approved by the dedicated approvers.
  1. The changes should be approved by the matched approvers that are configured in `OWNERS` files(layer by layer in folder level), and leaf approvers first.
  2. All the changes must be approved before merging.
4. `OWNERS` and `OWNERS_ALIASES` files should be updated on time.
  1. If file paths need to be updated, we should update the `OWNERS` files in the leaf folders.
  2. If the approver member should be updated, we should update the `/OWNERS_ALIASES` file.
  
## Role and Responsibility 

- For contributor
  - Assess the system's needs and requirements before making any configuration changes, considering factors such as performance, scalability, security, and compatibility. 
  - Implement the configuration changes and verify that any changes made to the codebase or configuration maintain backward compatibility.
  - Conduct testing to verify the effects of configuration changes, checking for any adverse impact on system behavior, performance, or compatibility.
  - In case of any issues or unexpected outcomes resulting from the configuration changes, contributors actively troubleshoot and work towards resolving them.
  - Document the configuration parameters that affect user behavior and any system variable changes (adding, deleting, deprecating, modifying) by **creating pull requests to documentation repositories** (`pingcap/docs`, `pingcap/docs-cn`), adding the changes to release notes, and updating relevant documentation, to ensure that users have access to accurate and up-to-date information.
- For approver
  1. Thoroughly review all proposed configuration change pull requests. This involves examining the details of the proposed modifications, such as the purpose, impact, and potential risks associated with the configuration change.
  2. Evaluate the impact these changes may have on the system or any related components. This includes considering potential risks, such as performance impact, security vulnerabilities, or conflicts with existing configurations.
  4. Validate that the proposed configuration changes in the pull requests have undergone appropriate testing.
  5. **Confirm the necessary documentation PR has been created.**
  6. Grant approval for the PR.

### PR DESCRIPTION
**Pull request by wuhuizuo**
### What is changed and how it works?

Issue Number: Close #17004

What's Changed:

```commit-message

```

### Related changes

- [ ] Need to cherry-pick to the release branch: release-8.1, release-7.5, release-7.1, release-6.5, release-6.1

### Check List

Tests 

- [x] No code

Side effects

- None

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by wuhuizuo** [comment]
/hold

Let @wuhuizuo to judge when to merge.

**Review by overvenus — COMMENTED** [review]
Please add `src/config/mod.rs` too.

**Comment by wuhuizuo** [comment]
> Please add `src/config/mod.rs` too.

done, PTAL

File: src/config/OWNERS
Comment (wuhuizuo) [inline src/config/OWNERS:7]: @overvenus Which module is this file primarily concerned with?

Comment (Connor1996) [inline src/config/OWNERS:7]: why transaction?

File: tests/integrations/config/OWNERS
Comment (Connor1996) [inline tests/integrations/config/OWNERS:7]: dito

File: src/storage/OWNERS
Comment (Connor1996) [inline src/storage/OWNERS:7]: dito

File: src/server/OWNERS
Comment (Connor1996) [inline src/server/OWNERS:7]: dito

File: src/config/OWNERS
Comment (overvenus) [reply src/config/OWNERS:7]: All modules, as it is the actual TiKV configuration struct.

**Comment by wuhuizuo** [comment]
PTAL @overvenus @Connor1996 

Also can you comment "/ok-to-test" when you send a LGTM? Thanks!

**Comment by wuhuizuo** [comment]
blocked by: https://github.com/dcoapp/app/issues/211

**Comment by wuhuizuo** [comment]
> > solved by workaround
> 
> Hey @wuhuizuo, I am curious if you have an OSS workaround for this issue.

@frouioui 
We switched to Prow's [DCO](https://github.com/kubernetes-sigs/prow/tree/main/pkg/plugins/dco) plugin temporarily. But this processing requires the premise that the organization's CI uses prow. 

Maybe you can try [deploy a private DCO instance](https://github.com/dcoapp/app/blob/main/docs/deploy.md). 

I saw you have solved it by another workaround.

File: tests/integrations/config/OWNERS
Comment (overvenus) [inline tests/integrations/config/OWNERS:5]: Can you skip this file? It's a test-only file, and it's a bit overkill to seek 'sig-critical-approvers-config-src' permission to add tests.

File: OWNERS_ALIASES
Comment (cfzjywxk) [inline OWNERS_ALIASES:10]: \+ zhangjinpeng87 
and 
+ [Connor1996](https://github.com/Connor1996)
+ [overvenus](https://github.com/overvenus)
I think

File: OWNERS_ALIASES
Comment (cfzjywxk) [inline OWNERS_ALIASES:6]: \+ [Connor1996](https://github.com/Connor1996)
\+ [overvenus](https://github.com/overvenus)
I think

File: OWNERS_ALIASES
Comment (cfzjywxk) [inline OWNERS_ALIASES:13]: \+ [overvenus](https://github.com/overvenus)
I think

**Review by cfzjywxk — COMMENTED** [review]
The two groups look fine to me, it could be splitted into more in the future if needed.

**Review by cfzjywxk — APPROVED** [review]

**Review by zhangjinpeng87 — APPROVED** [review]
LGTM

**Review by overvenus — APPROVED** [review]

**Review by Connor1996 — APPROVED** [review]

**Pull request merged**
